typescript/sdk/src/utils/sealevel/tokenSerialization.ts

```diff
@@ -0,0 +1,170 @@
+  /// The bump seed for this PDA.
+  bump!: number;
+  /// The address of the mailbox contract.
+  mailbox!: Uint8Array;
+  mailbox_pubkey!: PublicKey;
+  /// The Mailbox process authority specific to this program as the recipient.
+  mailbox_process_authority!: Uint8Array;
+  mailbox_process_authority_pubkey!: PublicKey;
+  /// The dispatch authority PDA's bump seed.
+  dispatch_authority_bump!: number;
+  /// The decimals of the local token.
+  decimals!: number;
+  /// The decimals of the remote token.
+  remote_decimals!: number;
+  /// Access control owner.
+  owner?: Uint8Array;
+  owner_pub_key?: PublicKey;
+  /// The interchain security module.
+  interchain_security_module?: Uint8Array;
+  interchain_security_module_pubkey?: PublicKey;
+  // The interchain gas paymaster
+  interchain_gas_paymaster?: {
+    address: Uint8Array;
+    type: number;
+  };
+  interchain_gas_paymaster_pubkey?: PublicKey;
+  // Gas amounts by destination
+  destination_gas?: Map<DomainId, bigint>;
+  /// Remote routers.
+  remote_routers?: Map<DomainId, Uint8Array>;
+  remote_router_pubkeys: Map<DomainId, PublicKey>;
+  constructor(public readonly fields: any) {
+    Object.assign(this, fields);
+    this.mailbox_pubkey = new PublicKey(this.mailbox);
+    this.mailbox_pubkey = new PublicKey(this.mailbox_process_authority);
+    this.owner_pub_key = this.owner ? new PublicKey(this.owner) : undefined;
+    this.interchain_security_module_pubkey = this.interchain_security_module
+      ? new PublicKey(this.interchain_security_module)
+      : undefined;
+    this.interchain_gas_paymaster_pubkey = this.interchain_gas_paymaster
+      ?.address
+      ? new PublicKey(this.interchain_gas_paymaster.address)
+      : undefined;
+    this.remote_router_pubkeys = new Map<number, PublicKey>();
+    if (this.remote_routers) {
+      for (const [k, v] of this.remote_routers.entries()) {
+        this.remote_router_pubkeys.set(k, new PublicKey(v));
+      }
+    }
+  }
+}
+
+export const HyperlaneTokenDataSchema = new Map<any, any>([
+  [
+    AccountDataWrapper,
+    {
+      kind: 'struct',
+      fields: [
+        ['initialized', 'u8'],
+        ['data', HyperlaneTokenData],
+      ],
+    },
+  ],
+  [
+    HyperlaneTokenData,
+    {
+      kind: 'struct',
+      fields: [
+        ['bump', 'u8'],
+        ['mailbox', [32]],
+        ['mailbox_process_authority', [32]],
+        ['dispatch_authority_bump', 'u8'],
+        ['decimals', 'u8'],
+        ['remote_decimals', 'u8'],
+        ['owner', { kind: 'option', type: [32] }],
+        ['interchain_security_module', { kind: 'option', type: [32] }],
+        [
+          'interchain_gas_paymaster',
+          {
+            kind: 'option',
+            type: {
+              kind: 'struct',
+              fields: [
+                ['address', [32]],
+                ['type', 'u8'],
+              ],
+            },
+          },
+        ],
+        // ['interchain_gas_paymaster_type', { kind: 'option', type: 'u8' }],
```

Comment:
delete this

Reply:
my bad, I commented that out but it should be there